dftimewolf/lib/exporters/timesketch.py

```diff
@@ -22,86 +27,130 @@ class TimesketchExporter(module.BaseModule):
+      message = 'Could not connect to Timesketch server'
       self.state.AddError(message, critical=True)
       return
 
-    if not self.sketch_id:  # No sketch id is provided, create it
+    if not self.sketch_id:
+      self.sketch_id = self._GetSketchIDFromAttributes()
+
+    if not self.sketch_id:  # No sketch id is provided, create it.
       if incident_id:
         sketch_name = 'Sketch for incident ID: ' + incident_id
       else:
         sketch_name = 'Untitled sketch'
       sketch_description = 'Sketch generated by dfTimewolf'
 
-      self.sketch_id = self.timesketch_api.CreateSketch(
+      sketch = self.timesketch_api.create_sketch(
           sketch_name, sketch_description)
+      self.sketch_id = sketch.id
       print('Sketch {0:d} created'.format(self.sketch_id))
 
+    if not sketch:
+      sketch = self.timesketch_api.get_sketch(self.sketch_id)
+
+    self.state.AddToCache('timesketch_sketch', sketch)
+    if analyzers and isinstance(analyzers, (tuple, list)):
+      self._analyzers = analyzers
+
+  def _GetSketchIDFromAttributes(self):
+    """Attempts to retrieve a Timesketch ID from ticket attributes.
+
+    Returns:
+      int: the sketch idenifier, or None if one was not available.
+    """
+    attributes = self.state.GetContainers(containers.TicketAttribute)
+    for attribute in attributes:
+      if attribute.name == self._SKETCH_ATTRIBUTE_NAME:
+        sketch_match = re.search(r'sketch/(\d+)/', attribute.value)
+        if sketch_match:
+          sketch_id = int(sketch_match.group(1), 10)
+          return sketch_id
+    return None
+
   def Process(self):
     """Executes a Timesketch export."""
-    # This is not the best way of catching errors, but timesketch_utils will be
-    # deprecated soon.
-    # TODO(tomchop): Consider using the official Timesketch python API.
-    if not self.timesketch_api.session:
+    if not self.timesketch_api:
       message = 'Could not connect to Timesketch server'
       self.state.AddError(message, critical=True)
 
-    named_timelines = []
-    for description, path in self.state.input:
-      if not description:
-        description = 'untitled timeline for '+path
-      named_timelines.append((description, path))
-    try:
-      self.timesketch_api.ExportArtifacts(named_timelines, self.sketch_id)
-    except RuntimeError as exception:
-      self.state.AddError(
-          'An error occurred while working with Timesketch: {0!s}'.format(
-              exception), critical=True)
-      return
+    sketch = self.state.GetFromCache('timesketch_sketch')
+    if not sketch:
+      sketch = self.timesketch_api.get_sketch(self.sketch_id)
 
-    sketch_url = self.timesketch_api.GetSketchUrl(self.sketch_id)
+    recipe_name = self.state.recipe.get('name', 'no_recipe')
+    input_names = []
+    for description, _ in self.state.input:
+      if not description:
+        continue
+      name, _, _ = description.rpartition('.')
```

Comment:
isn't `name = description.rpartition('.')[0]` a bit clearer?

Reply:
I'm not sure I agree with it being clearer, but nonetheless no strong feelings here so done....